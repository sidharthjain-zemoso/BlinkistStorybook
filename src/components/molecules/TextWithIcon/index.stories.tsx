import { ComponentStory, ComponentMeta } from "@storybook/react";

import IconText from "./index";

export default {
  title: "Molecules/Icon Text",
  component: IconText,
} as ComponentMeta<typeof IconText>;

const Template: ComponentStory<typeof IconText> = (args) => (
  <IconText {...args} />
);

export const iconText: ComponentStory<typeof IconText> = Template.bind({});
iconText.args = {
  icon: "user",
  iconAtEnd: true,
  text: "Some Text",
  iconHeight: "20px",
  iconWidth: "15px",
  textVariant: "subtitle1",
};
