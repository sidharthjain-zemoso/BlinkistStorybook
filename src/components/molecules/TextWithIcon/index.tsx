import Icon from "../../atoms/Icon";
import TypographyText from "../../atoms/Typography";

type TextWithIconProps = {
  iconHeight: string;
  iconWidth: string;
  textVariant:
    | "heading"
    | "heading3"
    | "subtitle1"
    | "subtitle2"
    | "subtitle3"
    | "body1"
    | "body2"
    | "body3"
    | "caption1"
    | "caption2"
    | "caption3";
  icon: string;
  iconAtEnd: boolean;
  text: string;
  className?: string;
  sx?: Object;
};

const TextWithIcon: React.FC<TextWithIconProps> = (props) => {
  return (
    <div
      data-testid="text-with-icon"
      style={{ display: "flex", padding: 5, alignItems: "center", ...props.sx }}
    >
      {!props.iconAtEnd ? (
        <Icon
          className={props.className}
          url={`/res/icons/${props.icon}.png`}
          sx={{ height: props.iconHeight, width: props.iconWidth }}
        />
      ) : null}
      <TypographyText
        className={props.className}
        component="div"
        variant={props.textVariant}
      >
        {props.text}
      </TypographyText>
      {props.iconAtEnd ? (
        <Icon
          className={props.className}
          url={`/res/icons/${props.icon}.png`}
          sx={{
            height: props.iconHeight,
            width: props.iconWidth,
            paddingLeft: 10,
          }}
        />
      ) : null}
    </div>
  );
};

export default TextWithIcon;
