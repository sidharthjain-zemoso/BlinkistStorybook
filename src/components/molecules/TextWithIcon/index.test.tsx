import { render, screen } from "@testing-library/react";
import TextWithIcon from ".";

describe("Text with Icon", () => {
  it("renders a text with icon with icon at end", () => {
    render(
      <TextWithIcon
        icon="user"
        iconAtEnd={true}
        text="Some Text"
        iconHeight="20px"
        iconWidth="15px"
        textVariant="subtitle1"
      />
    );

    const textElement = screen.getByText("Some Text");

    const iconElement = screen.getByRole("img");
    expect(textElement).toBeDefined();
    expect(iconElement).toBeDefined();

    const TextWithIconElement = screen.getByTestId("text-with-icon");
    expect(TextWithIconElement).toMatchSnapshot();
  });
  it("renders a text with icon with icon at start", () => {
    render(
      <TextWithIcon
        icon="user"
        iconAtEnd={false}
        text="Some Text"
        iconHeight="20px"
        iconWidth="15px"
        textVariant="subtitle1"
      />
    );

    const textElement = screen.getByText("Some Text");

    const iconElement = screen.getByRole("img");
    expect(textElement).toBeDefined();
    expect(iconElement).toBeDefined();

    const TextWithIconElement = screen.getByTestId("text-with-icon");
    expect(TextWithIconElement).toMatchSnapshot();
  });
});
