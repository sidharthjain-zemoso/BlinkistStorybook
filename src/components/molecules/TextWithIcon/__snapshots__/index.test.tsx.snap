// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text with Icon renders a text with icon with icon at end 1`] = `
<div
  data-testid="text-with-icon"
  style="display: flex; padding: 5px; align-items: center;"
>
  <div
    class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Some Text
  </div>
  <img
    alt="icon"
    src="/res/icons/user.png"
    style="margin-right: 5px; height: 20px; width: 15px; padding-left: 10px;"
  />
</div>
`;

exports[`Text with Icon renders a text with icon with icon at start 1`] = `
<div
  data-testid="text-with-icon"
  style="display: flex; padding: 5px; align-items: center;"
>
  <img
    alt="icon"
    src="/res/icons/user.png"
    style="margin-right: 5px; height: 20px; width: 15px;"
  />
  <div
    class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Some Text
  </div>
</div>
`;
